drools/src/main/java/com/datatorrent/drools/rules/listener/drool/RulesFiredListener.java

```diff
@@ -0,0 +1,136 @@
+/**
+ * Copyright (c) 2017 DataTorrent, Inc. ALL Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package com.datatorrent.drools.rules.listener.drool;
+
+import java.util.List;
+import java.util.Map;
+
+import org.kie.api.definition.rule.Rule;
+import org.kie.api.event.rule.AfterMatchFiredEvent;
+import org.kie.api.event.rule.AgendaEventListener;
+import org.kie.api.event.rule.AgendaGroupPoppedEvent;
+import org.kie.api.event.rule.AgendaGroupPushedEvent;
+import org.kie.api.event.rule.BeforeMatchFiredEvent;
+import org.kie.api.event.rule.MatchCancelledEvent;
+import org.kie.api.event.rule.MatchCreatedEvent;
+import org.kie.api.event.rule.RuleFlowGroupActivatedEvent;
+import org.kie.api.event.rule.RuleFlowGroupDeactivatedEvent;
+
+import org.apache.hadoop.classification.InterfaceStability;
+
+@InterfaceStability.Evolving
+
+/**
+ * Class implementing AgendaEventListener to track events when a Rule is matched and fired
+ */
+public class RulesFiredListener implements AgendaEventListener
+{
+
+  Map<Object, List<Rule>> factsAndFiredRules;
+  Map<Rule, Integer> ruleCount;
+
+  public RulesFiredListener(Map<Object, List<Rule>> factsAndFiredRules, Map<Rule, Integer> ruleCount)
```

Comment:
Ok then I can close this PR and create a new one once your operator template is merged.